It doesn't compile on FreeBSD

% freebsd-version
11.1-RELEASE-p6
ferz@w1:/tmp/mkvserver_mk2 % gmake
cc -g -Wall -I/usr/local/include -L/usr/local/lib -lavformat -lavcodec -lavutil  -lpthread -c segment.c
cc: warning: -lavformat: 'linker' input unused [-Wunused-command-line-argument]
cc: warning: -lavcodec: 'linker' input unused [-Wunused-command-line-argument]
cc: warning: -lavutil: 'linker' input unused [-Wunused-command-line-argument]
cc: warning: -lpthread: 'linker' input unused [-Wunused-command-line-argument]
cc: warning: argument unused during compilation: '-L/usr/local/lib' [-Wunused-command-line-argument]
In file included from segment.c:2:
./segment.h:25:5: error: unknown type name 'pthread_mutex_t'
    pthread_mutex_t nb_read_lock;
    ^
1 error generated.
gmake: *** [Makefile:9: segment.o] Error 1

--- a/Makefile
+++ b/Makefile
@@ -1,19 +1,23 @@
+CC = cc
+
+LAV_LDFLAGS = `pkg-config --libs libavformat libavcodec libavutil`
+LAV_CFLAGS = `pkg-config --cflags libavformat libavcodec libavutil`
+
+.PHONY: all clean
+
 all: server
-LAV_FLAGS = $(shell pkg-config --libs --cflags libavformat libavcodec libavutil)
-# LAV_FLAGS = -L/usr/local/lib -lavcodec -lavformat -lavutil
 
 server: segment.o buffer.o publisher.o server2.c
-	cc -g -Wall $(LAV_FLAGS)  -lpthread -o server segment.o buffer.o publisher.o server2.c
+	$(CC) -g -Wall $(LAV_CFLAGS) $(LAV_LDFLAGS) -pthread -o server segment.o buffer.o publisher.o server2.c
 
 segment.o: segment.c segment.h
-	cc -g -Wall $(LAV_FLAGS) -lpthread -c segment.c
+	$(CC) -g -Wall $(LAV_CFLAGS) -pthread -c segment.c
 
 buffer.o: buffer.c buffer.h
-	cc -g -Wall $(LAV_FLAGS) -lpthread -c buffer.c
+	$(CC) -g -Wall $(LAV_CFLAGS) -pthread -c buffer.c
 
 publisher.o: publisher.c publisher.h
-	cc -g -Wall $(LAV_FLAGS) -lpthread -c publisher.c
-
+	$(CC) -g -Wall $(LAV_CFLAGS) -pthread -c publisher.c
 
 clean:
 	rm *.o server

--- a/segment.h
+++ b/segment.h
@@ -2,6 +2,7 @@
 #define SEGMENT_H
 
 #include <libavformat/avformat.h>
+#include <pthread.h>
 
 #define AV_BUFSIZE 4096
 

